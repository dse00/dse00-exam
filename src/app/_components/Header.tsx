import Link from 'next/link';
import { FC } from 'react';

import LoginDialog from '@/components/LoginDialog';

import AvatarAndMenu from './AvatarAndMenu';
import HeaderIcon from './HeaderIcon';
import HeaderScoreChartButton from './HeaderScoreChartButton';
import MessageDialog from './MessageDialog';

const Header: FC = () => {
  return (
    <>
      <header className='bg-main w-full h-[58px] flex justify-center items-center px-2'>
        <div className='container flex items-center sm:gap-10 gap-3'>
          <HeaderIcon />
          {/* {isExamPage ? (
          <div className='flex items-center gap-3'>
            <Image src='/images/leaf.png' width={34} height={34} alt='icon' />
            <ProblemListSidebar />
          </div>
        ) : (
          <Link href={'/'} className='text-white font-black text-4xl drop-shadow-[0_0px_6px_rgba(255,255,255,0.7)]'>
            DSE00 exam
          </Link>
        )} */}

          <div className='flex items-center gap-10 grow'>
            <div className='sm:flex gap-10 hidden'>
              <Link
                className={'text-white text-[15px]'}
                href={'https://www.dse00.com/p/core-cutoff.html'}
                target='_blank'
              >
                2024 Cut-off
              </Link>
              <Link className={'text-white text-[15px]'} href={'https://www.dse00.com/p/jupas.html'} target='_blank'>
                JUPAS Cuf-off
              </Link>
            </div>
          </div>
          <HeaderScoreChartButton />
          <AvatarAndMenu />
          <MessageDialog />
        </div>
      </header>
      <LoginDialog />
    </>
  );
};

export default Header;
