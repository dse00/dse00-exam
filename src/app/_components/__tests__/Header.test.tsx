import '@testing-library/jest-dom';

import { render } from '@testing-library/react';

import Header from '../Header';

// Mock the AvatarAndMenu component
jest.mock('../AvatarAndMenu', () => () => <div>Mocked AvatarAndMenu</div>);

describe('Header Component', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
