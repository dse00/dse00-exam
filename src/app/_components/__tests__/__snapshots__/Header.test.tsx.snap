// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-main w-full h-[58px] flex justify-center items-center px-2"
  >
    <div
      class="container flex items-center justify-between"
    >
      <div
        class="flex items-center gap-10"
      >
        <a
          class="text-white font-black text-4xl drop-shadow-[0_0px_6px_rgba(255,255,255,0.7)]"
          href="/"
        >
          DSE00 exam
        </a>
        <div
          class="sm:flex gap-5 hidden"
        >
          <a
            class="text-white text-[15px]"
            href="https://www.dse00.com/p/core-cutoff.html"
            target="_blank"
          >
            DSE00 AI 補習配對
          </a>
          <a
            class="text-white text-[15px]"
            href="https://www.dse00.com/p/core-cutoff.html"
            target="_blank"
          >
            2024 Cut-off
          </a>
          <a
            class="text-white text-[15px]"
            href="https://www.dse00.com/p/core-cutoff.html"
            target="_blank"
          >
            JUPAS Cuf-off
          </a>
        </div>
      </div>
      <div>
        Mocked AvatarAndMenu
      </div>
    </div>
  </header>
</DocumentFragment>
`;
